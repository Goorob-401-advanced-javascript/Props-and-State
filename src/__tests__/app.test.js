/* eslint-disable no-unused-vars */
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Form from '../components/form.js';

Enzyme.configure({adapter: new Adapter() });

describe('<Form />', ()=>{
    it('if it exsit or not ',()=>{
        let app = shallow(<Form />);
        expect(app.find('form').exists()).toBeTruthy();
        expect(app.find('button').exists()).toBeTruthy();
    
      });
  it('render Form',()=>{
    let app = mount(<Form/>);
    let form = app.find('form');
    expect(form.exists()).toBeTruthy();
  });

  it('can render to the DOM',()=>{
    const rendered = renderer.create(<Form />).toJSON();
    expect(rendered).toMatchSnapshot();
  });


});