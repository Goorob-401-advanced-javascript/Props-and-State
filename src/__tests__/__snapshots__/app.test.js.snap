// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> can render to the DOM 1`] = `
<main>
  <section>
    <form
      onSubmit={[Function]}
    >
      <section>
        <input
          className="url"
          name="url"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          className="methods"
        >
          <label>
            <input
              name="method"
              type="radio"
              value="get"
            />
            <span>
              GET
            </span>
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="post"
            />
            <span>
              POST
            </span>
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="put"
            />
            <span>
              PUT
            </span>
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="delete"
            />
            <span>
              DELETE
            </span>
          </label>
          <label>
             
            <button
              type="submit"
            >
               GO  ! 
            </button>
             
          </label>
        </div>
      </section>
    </form>
  </section>
</main>
`;
